{"name": "Twitter link", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: stale element not found in the current frame\n  (Session info: chrome=132.0.6834.83); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception\nStacktrace:\n#0 0x562d4ee737ca <unknown>\n#1 0x562d4e96b2f0 <unknown>\n#2 0x562d4e9712f1 <unknown>\n#3 0x562d4e973988 <unknown>\n#4 0x562d4e973a13 <unknown>\n#5 0x562d4e9b5608 <unknown>\n#6 0x562d4e9df9b2 <unknown>\n#7 0x562d4e9aeb78 <unknown>\n#8 0x562d4e9dfb7e <unknown>\n#9 0x562d4e9fe3b3 <unknown>\n#10 0x562d4e9df753 <unknown>\n#11 0x562d4e9acbaa <unknown>\n#12 0x562d4e9addfe <unknown>\n#13 0x562d4ee3e38b <unknown>\n#14 0x562d4ee42307 <unknown>\n#15 0x562d4ee2ae7c <unknown>\n#16 0x562d4ee42ec7 <unknown>\n#17 0x562d4ee0f24f <unknown>\n#18 0x562d4ee622f8 <unknown>\n#19 0x562d4ee624c0 <unknown>\n#20 0x562d4ee72646 <unknown>\n#21 0x7f2b2a49ca94 <unknown>\n#22 0x7f2b2a529c3c <unknown>", "trace": "self = <Tests.test_SocialMediaLinks.TestLogin object at 0x7f9261e01fd0>\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"ab4cbe977a6db088d500bda6d6d9a668\")>\n\n    @severity(severity_level.NORMAL)\n    @allure.feature('Social Media Links')\n    @allure.title('Twitter link')\n    def test_1(self, driver):\n        landing_page_obj = LandingPage(driver)\n        home_page_obj = HomePage(driver)\n        driver.get(BaseClass.url)\n        landing_page_obj.complete_login(BaseClass.username, BaseClass.password)\n        home_page_obj.click_twitter_button()\n        with check, allure.step(\"Check twitter heather tex\"):\n            driver.switch_to.window(driver.window_handles[1])\n>           assert home_page_obj.get_twitter_heather_text() == \"Sauce Labs\"\n\nTests/test_SocialMediaLinks.py:25: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \nPages/HomePage.py:46: in get_twitter_heather_text\n    return twitter_heather.text\n/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/selenium/webdriver/remote/webelement.py:90: in text\n    return self._execute(Command.GET_ELEMENT_TEXT)[\"value\"]\n/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/selenium/webdriver/remote/webelement.py:395: in _execute\n    return self._parent.execute(command, params)\n/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py:347: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x7f9261d3d990>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"stale element reference\",\"message\":\"stale element reference: stale elemen...\\\\n#20 0x562d4ee72646 \\\\u003Cunknown>\\\\n#21 0x7f2b2a49ca94 \\\\u003Cunknown>\\\\n#22 0x7f2b2a529c3c \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: stale element not found in the current frame\nE         (Session info: chrome=132.0.6834.83); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception\nE       Stacktrace:\nE       #0 0x562d4ee737ca <unknown>\nE       #1 0x562d4e96b2f0 <unknown>\nE       #2 0x562d4e9712f1 <unknown>\nE       #3 0x562d4e973988 <unknown>\nE       #4 0x562d4e973a13 <unknown>\nE       #5 0x562d4e9b5608 <unknown>\nE       #6 0x562d4e9df9b2 <unknown>\nE       #7 0x562d4e9aeb78 <unknown>\nE       #8 0x562d4e9dfb7e <unknown>\nE       #9 0x562d4e9fe3b3 <unknown>\nE       #10 0x562d4e9df753 <unknown>\nE       #11 0x562d4e9acbaa <unknown>\nE       #12 0x562d4e9addfe <unknown>\nE       #13 0x562d4ee3e38b <unknown>\nE       #14 0x562d4ee42307 <unknown>\nE       #15 0x562d4ee2ae7c <unknown>\nE       #16 0x562d4ee42ec7 <unknown>\nE       #17 0x562d4ee0f24f <unknown>\nE       #18 0x562d4ee622f8 <unknown>\nE       #19 0x562d4ee624c0 <unknown>\nE       #20 0x562d4ee72646 <unknown>\nE       #21 0x7f2b2a49ca94 <unknown>\nE       #22 0x7f2b2a529c3c <unknown>\n\n/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py:229: StaleElementReferenceException"}, "steps": [{"name": "Check twitter heather tex", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: stale element not found in the current frame\n  (Session info: chrome=132.0.6834.83); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception\nStacktrace:\n#0 0x562d4ee737ca <unknown>\n#1 0x562d4e96b2f0 <unknown>\n#2 0x562d4e9712f1 <unknown>\n#3 0x562d4e973988 <unknown>\n#4 0x562d4e973a13 <unknown>\n#5 0x562d4e9b5608 <unknown>\n#6 0x562d4e9df9b2 <unknown>\n#7 0x562d4e9aeb78 <unknown>\n#8 0x562d4e9dfb7e <unknown>\n#9 0x562d4e9fe3b3 <unknown>\n#10 0x562d4e9df753 <unknown>\n#11 0x562d4e9acbaa <unknown>\n#12 0x562d4e9addfe <unknown>\n#13 0x562d4ee3e38b <unknown>\n#14 0x562d4ee42307 <unknown>\n#15 0x562d4ee2ae7c <unknown>\n#16 0x562d4ee42ec7 <unknown>\n#17 0x562d4ee0f24f <unknown>\n#18 0x562d4ee622f8 <unknown>\n#19 0x562d4ee624c0 <unknown>\n#20 0x562d4ee72646 <unknown>\n#21 0x7f2b2a49ca94 <unknown>\n#22 0x7f2b2a529c3c <unknown>\n\n", "trace": "  File \"/home/runner/work/new_demo_project/new_demo_project/Tests/test_SocialMediaLinks.py\", line 25, in test_1\n    assert home_page_obj.get_twitter_heather_text() == \"Sauce Labs\"\n  File \"/home/runner/work/new_demo_project/new_demo_project/Pages/HomePage.py\", line 46, in get_twitter_heather_text\n    return twitter_heather.text\n           ^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/selenium/webdriver/remote/webelement.py\", line 90, in text\n    return self._execute(Command.GET_ELEMENT_TEXT)[\"value\"]\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/selenium/webdriver/remote/webelement.py\", line 395, in _execute\n    return self._parent.execute(command, params)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py\", line 347, in execute\n    self.error_handler.check_response(response)\n  File \"/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "start": 1737859322785, "stop": 1737859324547}], "parameters": [{"name": "driver", "value": "'Chrome'"}], "start": 1737859322317, "stop": 1737859324549, "uuid": "4e58c9da-3d4f-4283-a925-7ed3d9fc484b", "historyId": "f3d4fdd619ae32f73e46d5a7dc8251a4", "testCaseId": "101d61ff090e1c199fa44f306a2af274", "fullName": "Tests.test_SocialMediaLinks.TestLogin#test_1", "labels": [{"name": "feature", "value": "Social Media Links"}, {"name": "severity", "value": "normal"}, {"name": "parentSuite", "value": "Tests"}, {"name": "suite", "value": "test_SocialMediaLinks"}, {"name": "subSuite", "value": "TestLogin"}, {"name": "host", "value": "fv-az1391-799"}, {"name": "thread", "value": "2019-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "Tests.test_SocialMediaLinks"}]}